modules/nix/linux-builder.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (pkgs) stdenv;
+
+  cfg = config.nix.linux-builder;
+in
+
+{
+  options.nix.linux-builder = {
+    enable = mkEnableOption "Linux builder";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.darwin.builder;
+      defaultText = "pkgs.darwin.builder";
+      example = literalExpression "pkgs.darwin.builder.override { modules = [ { virtualisation.darwin-builder.hostPort = 22; } ] }";
+      description = ''
+        This option specifies the Linux builder to use.
+
+        You can add extra NixOS modules and configuration to the builder by overriding `modules`. You should be
+        running the Linux builder without any overrides (by not setting this option) before changing this option.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [ {
+      assertion = config.nix.settings.trusted-users != [ "root" ] || config.nix.settings.extra-trusted-users != [ ];
+      message = ''
+        Your user or group (@admin) needs to be added to `nix.settings.trusted-users` or `nix.settings.extra-trusted-users`
+        to use the Linux builder.
+      '';
+    } ];
+
+    system.activationScripts.preActivation.text = ''
+      mkdir -p /var/lib/darwin-builder
+    '';
+
+    launchd.daemons.linux-builder = {
+      environment = {
+        inherit (config.environment.variables) NIX_SSL_CERT_FILE;
+      };
+      serviceConfig = {
+        ProgramArguments = [
+          "/bin/sh" "-c"
+          "/bin/wait4path /nix/store &amp;&amp; exec ${cfg.package}/bin/create-builder"
+        ];
+        KeepAlive = true;
+        RunAtLoad = true;
+        WorkingDirectory = "/var/lib/darwin-builder";
+        StandardOutPath = "/var/log/darwin-builder.log";
```

Comment:
Launchd doesn't have any log rotation so I would avoid enabling this by default in modules, a use can always enable logs for debugging in the local configuration. Or dynamically attach the log output as described here https://github.com/LnL7/nix-darwin/blob/master/modules/services/nix-daemon.nix#L28-L31.

Maybe the description of the Standard*Path options should also be extended with some information about that.

Reply:
Removed :+1: